### DISCUSSION TITLE
yasa.sleep_statistics

### DISCUSSION THREAD

**Discussion by olly1987** [raphaelvallat/yasa#148] [Q&A] [answered]
Hi Raphael, thanks for your amazing work with YASA!
I have a couple of questions about how the sleep statistics are calculated.
I read here (https://raphaelvallat.com/yasa/build/html/generated/yasa.sleep_statistics.html#yasa.sleep_statistics) that YASA follows the AASM guidelines to calculate the sleep parameters. However, could you please specify how the following three parameters are calculated?

**Sleep onset latency**: Latency to first epoch of any sleep --> my question is: is one epoch enough, or do you have a condition like the first epoch of sleep (for example N1) needs to be followed by at least 2 more consecutive N1 epochs or by any other sleep stage (for example N2? So if you have W-W-N1-W-N1-N2 is sleep onset the first N1 epoch or the second one?

**Sleep Period Time**: duration from first to last period of sleep --> my question here is similar to above: is the fist period of sleep the first epoch of any sleep stage independent of what follows or is there a conditional rule? For example 3 consecutive N1 epochs are needed to classify this as first period of sleep. 

**Latencies**: latencies of sleep stages from the beginning of the record --> my question is: REM latency should be the time from sleep onset to first epoch of REM according to the AASM guidelines. Is REM latency calculated like this, or as time from the beginning of the record? 

Thanks a lot!

**Comment by raphaelvallat** [comment, marked as answer]
Hi @olly1987,

**Sleep onset latency**: no condition, SOL is simply the latency to the first epoch of sleep. We could implement a "latency to persistent sleep" if desired. 

**Sleep Period Time**:

>  is the fist period of sleep the first epoch of any sleep stage independent of what follows

Correct

**Latencies**

> REM latency should be the time from sleep onset to first epoch of REM according to the AASM guidelines.

As of now, "Lat_REM" returned by YASA is the latency from the beginning of the record to first epoch of sleep, which is consistent with other sleep stages. You can easily calculate the AASM REM latency by substracting the SOL. However, you are right that this should be made more explicit in the documentation of the function. I'll open an issue now.

Hope this helps!

**Reply by olly1987** [reply]
Hi Raphael, thanks for the reply! This is very helpful. 
I think adding a latency to persistent sleep would be good. I know that AASM defines sleep latency as time to the first epoch of any sleep but I personally do not agree with this definition. Or better I think there is a big conceptual difference between the latency to the first epoch of any sleep (e.g. N1), independent of what follows, which in my data is often again wakefulness, and the latency to persistent sleep (e.g. 3 consecutive N1 epochs or the first N2 epoch. If I am not mistaken these were the old R&K rules). 

How can I be notify if you decide to add the parameter "latency to persistent sleep"? Many thanks again!

**Reply by raphaelvallat** [reply]
Thanks @olly1987 , I agree. You can be notified of new releases by "Watching" the GitHub repo:

<img width="1398" alt="image" src="https://github.com/raphaelvallat/yasa/assets/25703792/2a382275-0be9-457f-b00e-bc253235a903">
